### ISSUE TITLE
Array declaration sniff should account for places where WP expects mixed key/no-key arrays

### ISSUE THREAD

**Issue by JDGrimes** [WordPress-Coding-Standards/WordPress-Coding-Standards#146]
The array syntax sniff will report when an array has some values assigned specific keys and others not.

> 314 | WARNING | No key specified for array entry; first entry specifies key
>     |         | (WordPress.Arrays.ArrayDeclaration)

This causes a problem where WordPress requires mixed arrays, as in the `'meta_query'` argument for [`WP_Query`](http://codex.wordpress.org/Class_Reference/WP_Query#Custom_Field_Parameters):

``` php
$args = array(
    'post_type' => 'product',
    'meta_query' => array(
        'relation' => 'OR', // 'relation' key used to specify AND/OR, the rest are numeric
        array(
            'key' => 'color',
            'value' => 'blue',
            'compare' => 'NOT LIKE'
        ),
        array(
            'key' => 'price',
            'value' => array( 20, 100 ),
            'type' => 'numeric',
            'compare' => 'BETWEEN'
        )
    )
);
$query = new WP_Query( $args );
```

The sniff should probably ignore any key if its text is `'related'`.
